core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteWithDeletionVectorsHelper.scala

```diff
@@ -341,11 +342,16 @@ object DeletionVectorBitmapGenerator {
     : Seq[DeletionVectorResult] = {
     // TODO: fix this to work regardless of whether Spark encodes or doesn't encode
     //  _metadata.file_path. See https://github.com/delta-io/delta/issues/1725
-    val uriEncode = DeltaUDF.stringFromString(path => {
-      new Path(path).toUri.toString
-    })
+    // Build two maps, using Path or String as keys. The one with String keys is used in UDF.
+    val canonicalizedPathMap = buildCanonicalizedPathMap(txn.deltaLog, candidateFiles)
+    val canonicalizedPathStringMap =
+      canonicalizedPathMap.map { case (k, v) => k.toString -> v }
```

Comment:
Why have this second pass, for a single-callsite helper method? Can it just return a string-string map directly, since that's what we ultimately broadcast?

Reply:
My concern is Path.toString being slow. When the log is huge it might have some slowdowns.

Reply:
Thanks for the comment! I addressed your comment in a later PR https://github.com/delta-io/delta/pull/1770.